app/timetables/admin.py

```diff
@@ -20,18 +25,24 @@ class AdminsInline(admin.TabularInline):
 
 
 @admin.register(Timetable)
-class TimetableAdmin(DefaultAdmin):
+class TimetableAdmin(CommonReadOnlyFieldsAdmin):
     fields = ('name', 'slug', 'code', 'api_key', 'cycle_length',
-              'current_cycle_day', 'description')
+              'current_cycle_day', 'description', 'date_created', 'date_modified')
     inlines = (AdminsInline,)
 
 
 @admin.register(Dish)
-class DishAdmin(DefaultAdmin):
-    fields = ('name', 'slug', 'description')
+class DishAdmin(CommonReadOnlyFieldsAdmin):
+    fields = ('name', 'slug', 'description', 'date_created', 'date_modified')
+
+
+@admin.register(MenuItem)
+class MenuItemAdmin(admin.ModelAdmin):
+    readonly_fields = ('date_created', 'date_modified')
+    fields = ('timetable', 'cycle_day', 'meal', 'meal_option', 'date_created', 'date_modified')
 
```

Comment:
@andela-kerinoso the `CommonReadOnlyFieldsAdmin` was reused by two classes. Maybe I should have called it something like `SlugTimeStampReadOnlyFieldsAdmin` since it is for slug and read-only fields from TimeStampMixin instead of repeating same line over and over again by explicitly stating the fields for each Admin class.